genai-perf/genai_perf/profile_data_parser/llm_profile_data_parser.py

```diff
@@ -182,26 +183,44 @@ def _preprocess_response(
             # Sometimes streamed chunks are returned in a splintered fashion.
             # This forces a merge with the previous chunk if error detected.
             if len(res_outputs) > 1:
-                for i in reversed(range(len(res_outputs))):
+                for i in reversed(range(1, len(res_outputs))):
                     response = res_outputs[i]["response"]
+
+                    # Ignore keepalive SSE response
+                    if response.strip() == ":":
+                        res_outputs[i]["response"] = ""
+                        continue
```

Comment:
Probably not intended side affect but I don't think you even need to change this part, you can keep the logic of stitching chunks until "data:" is found and it should still work in combination of the suggested multi-response change below..
i.e. if you have two chunk `"data: {}\n\n"`, `":\n\n"`, existing logic will merge them to `"data: {}\n\n:\n\n"`, and later multi-response processing will split it back to `["data: {}", ":"]` and only `"data: {}"` will remain after the filtering.